ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -0,0 +1,565 @@
+layer to use also the data from the same deployment to create this product), or
+as an alternative, the script can be run only with a deployment code string value
+
+Please refer to ./anmn_temp_gridded_product.py -h for more description related to
+inputs
+
+The script will return path to the new FV02 product and the relative path of the
+FV02 product already on the data storage if exists (which needs to be removed)
+respectively
+
+Author: laurent.besnard@utas.edu.au
+"""
+
+
+from datetime import datetime, timedelta
+from generate_netcdf_att import generate_netcdf_att, get_imos_parameter_info
+from imos_logging import IMOSLogging
+from matplotlib import gridspec
+from netCDF4 import Dataset, num2date, date2num
+from util import get_git_revision_script_url
+from util import wfs_request_matching_file_pattern
+import argparse
+import bisect
+import numpy as np
+import os
+import pandas as pd
+import pylab as pl
+import re
+import sys
+import shutil
+import tempfile
+import urllib2
+
+
+def plot_abs_comparaison_old_new_product(old_product_rel_path, new_nc_path):
+    """
+    create optional plots between old product if exists and new one.
+    """
+    tmp_old_prod_path = download_list_nc([os.path.join(s3_bucket_prefix, old_product_rel_path)])
+    old_nc_path = os.path.join(tmp_old_prod_path, os.path.basename(old_product_rel_path))
+
+    nc_old_obj = Dataset(old_nc_path, 'r')
+    nc_new_obj = Dataset(new_nc_path, 'r')
+
+    d_index_eq = []
+    for d_value in nc_new_obj['DEPTH']:
+        d_index_eq.append(find_closest(nc_old_obj['DEPTH'][:], d_value))
+
+    t_index_eq = (find_closest(nc_old_obj['TIME'][:], nc_new_obj['TIME'][0]))
+
+    diff_mean_temp_per_depth = []
+    for idx, d in enumerate(d_index_eq):
+        diff_mean_temp_per_depth.append(np.nanmean(nc_new_obj['TEMP'][:, idx]) - np.nanmean(nc_old_obj['TEMP'][:, d]))
+
+    if len(nc_old_obj['TEMP'][t_index_eq:]) == len(nc_new_obj['TEMP']) :
+        df = (nc_new_obj['TEMP'][:, :] - nc_old_obj['TEMP'][t_index_eq:, d_index_eq, 0, 0])
+    elif len(nc_old_obj['TEMP'][t_index_eq:]) < len(nc_new_obj['TEMP']):
+        max_len = len(nc_old_obj['TEMP'][t_index_eq:])
+        df = (nc_new_obj['TEMP'][:max_len, :] - nc_old_obj['TEMP'][t_index_eq:, d_index_eq, 0, 0])
+    elif len(nc_old_obj['TEMP'][t_index_eq:]) > len(nc_new_obj['TEMP']):
+        max_len = len(nc_new_obj['TEMP'][t_index_eq:])
+        df = (nc_new_obj['TEMP'][:max_len, :] - nc_old_obj['TEMP'][:max_len, d_index_eq, 0, 0])
+
+    df = pd.DataFrame(df)
+    df = df.transpose()
+
+    x = df.columns.values
+    y = df.index.values
+    Z = df.values
+
+    fig = pl.figure(figsize=(30, 30))
+    gs  = gridspec.GridSpec(5, 5)
+
+    # ax1
+    ax1 = fig.add_subplot(gs[:, 0:4])
+    pcm = ax1.contourf(x, y, Z, 50, vmin=-np.nanmax(Z), cmap=pl.cm.RdBu_r)
+    fig.gca().invert_yaxis()
+    fig.colorbar(pcm, ax=ax1, extend='both', orientation='vertical')
+    ax1.set_ylabel('Depth in meters')
```

Comment:
Values on the depth axis can be misinterpreted as actual depth, while it is in fact the index in the depth vector.